unlock-app/src/services/web3Service.js

```diff
@@ -141,61 +154,50 @@ export default class Web3Service {
 
   /**
    * Creates a lock on behalf of the user `from`.
-   * This returns a promise of lock, but also accepts a callback to monitor the transaction
    * @param {PropTypes.lock} lock
-   * @param {function} callback
-   * @return Promise<Lock>
    */
-  createLock(lock, owner, callback) {
-    return new Promise((resolve, reject) => {
-      const unlock = new this.web3.eth.Contract(UnlockContract.abi, UnlockContract.networks[this.networkId].address)
-
-      const data = unlock.methods.createLock(
-        lock.expirationDuration,
-        lock.keyPrice,
-        lock.maxNumberOfKeys
-      ).encodeABI()
-
-      // The transaction object!
-      const transaction = {
-        status: 'pending',
-        confirmations: 0,
-        createdAt: new Date().getTime(),
-        lock: lock.id,
-      }
+  createLock(lock, owner) {
+    const unlock = new this.web3.eth.Contract(UnlockContract.abi, this.unlockContractAddress)
+
+    const data = unlock.methods.createLock(
+      lock.expirationDuration,
+      lock.keyPrice,
+      lock.maxNumberOfKeys
+    ).encodeABI()
+
+    // The transaction object!
+    const transaction = {
+      status: 'pending',
+      confirmations: 0,
+      createdAt: new Date().getTime(),
+      lock: lock.id,
+    }
 
-      return this.sendTransaction({
-        to: UnlockContract.networks[this.networkId].address,
-        from: owner.address,
-        data: data,
-        gas: 2000000,
-        privateKey: owner.privateKey,
-        contractAbi: UnlockContract.abi,
-      }, (error, { event, args }) => {
-        if (error) {
-          console.error(error)
-        }
-        if (event === 'transactionHash') {
-          transaction.hash = args.hash
-          transaction.status = 'submitted'
-          lock.transaction = transaction.hash
-          callback(transaction, lock)
-        } else if (event === 'confirmation') {
-          transaction.status = 'mined'
-          transaction.confirmations += 1
-          callback(transaction, lock)
-        } else if (event === 'NewLock') {
-          lock.address = args.newLockAddress
-          return this.refreshLock(lock)
-            .then((lock) => {
-              callback(transaction, lock)
-              return resolve(lock)
-            })
-        }
-      }).catch((error) => {
-        console.error(error)
-        return reject(error)
-      })
+    return this.sendTransaction({
+      to: this.unlockContractAddress,
+      from: owner.address,
+      data: data,
+      gas: 2000000,
+      contractAbi: UnlockContract.abi,
+    }, (error, { event, args } = {}) => {
+      if (error) {
+        this.emit('error', error)
+      }
+      if (event === 'transactionHash') {
+        transaction.hash = args.hash
+        transaction.status = 'submitted'
+        lock.transaction = transaction.hash
```

Comment:
I kind of wish there was a more syntactically elegant way to do this - maybe a helper function that applies the properties of a new object given as a parameter to an existing object?

Reply:
Yeah, agreed. I think we need to think about  our "models" more!